### PR TITLE
Upkeep - drop crayon, drop mockr

### PR DESCRIPTION
**Pull request by olivroy**
* Drop crayon 
* Drop mockr to use `local_mocked_bindings()`
* Remove home-grown `expect_no_warning()`
* Test lints (`expect_null()`, `expect_match()`, etc.

Conflicts still print as expected

image

### CODE REVIEW

File: tests/testthat/test-dll.R [hunk 1]
```diff
@@ -45,32 +44,32 @@ test_that("load_all() compiles and loads DLLs", {
 
   # Check that it's loaded properly, by running a function from the package.
   # nulltest() calls a C function which returns null.
-  expect_true(is.null(nulltest()))
+  expect_null(nulltest())
 
   # DLL should be listed in .dynLibs()
   dynlibs <- vapply(.dynLibs(), `[[`, "name", FUN.VALUE = character(1))
-  expect_true(any(grepl("testDllLoad", dynlibs)))
```
Comment (lionel-) [inline tests/testthat/test-dll.R:52]: Thanks, I just have a question here, why was this removed?

File: tests/testthat/test-dll.R [hunk 2]
```diff
@@ -45,32 +44,32 @@ test_that("load_all() compiles and loads DLLs", {
...
+  pkgbuild::clean_dll("testDllLoad")
```
Comment (olivroy) [inline tests/testthat/test-dll.R:51]:
```suggestion
  expect_match(dynlibs, "TestDllLoad", all = FALSE)
  pkgbuild::clean_dll("testDllLoad")
```

**Comment by lionel-** [comment]
Thanks!
